sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -812,7 +812,11 @@ class Analyzer(
 
       case i @ InsertIntoStatement(u: UnresolvedRelation, _, _, _, _) if i.query.resolved =>
         lookupV2Relation(u.multipartIdentifier)
-          .map(v2Relation => i.copy(table = v2Relation))
+          .map {
+            EliminateSubqueryAliases(_) match {
```

Comment:
is it still needed? `lookupV2Relation` doesn't add `SubqueryAlias`

Reply:
ah it hasn't been changed yet. I think it's better to let `CatalogV2Util.loadRelation` still return `DataSourceV2Relation`, and we add `SubqueryAlias` in the analyzer explicitly.